### PR TITLE
*: clean up and normalize German translation

### PR DESCRIPTION
**Pull request by marchersimon**
The german pages were completely inconsistent and often full of grammar and spelling mistakes and bad translations. They're still not perfect yet, but it's a step in the right direction. 
Also, during the process I had some ideas that could be of interest:

- **Language-specific tranlsation guides**
I thought of writing a small guide for each language with best practices and things to avoid. That file could be located in each languages folder. A small file with best practices and things to avoid would really help keep the pages consistent.

- **Page versioning system**
This one's probably too hard to implement, but however. If each page had a version number, which gets increased by one, each time it is edited, it would be so easy to see which translations are outdated. Otherwise the only possibility we have is to manually compare the pages.

- **Pages with same name**
Lastly, I saw that there is a `cd.md` in `common` and one in `windows`, which really are different commands. `tldr cd` is completely ignoring the windows one. Is there a system to avoid this?

### CODE REVIEW

**Comment by bl-ue** [comment]
Wow @marchersimon this is so amazing—I don't think we've ever had big rewrite like this ever! 🤩 

Is it a WIP?

**Comment by bl-ue** [comment]
>Lastly, I saw that there is a cd.md in common and one in windows, which really are different commands. tldr cd is completely ignoring the windows one. Is there a system to avoid this?

Yes, there is.  Pages in the platform specific folders take precedence over the ones in `common/`.  What client/version are you using?

**Comment by marchersimon** [comment]
> Wow @marchersimon this is so amazing—I don't think we've ever had big rewrite like this ever! :star_struck:

Honestly, I didn't think I'd make them all, and I was afraid it's to much for one PR to review :sweat_smile: 

> Is it a WIP?

Nope, I'm pretty much done. 

>  Pages in the platform specific folders take precedence over the ones in `common/`.

Well, that makes sense. I really didn't think of that... :smile:

**Comment by bl-ue** [comment]
>Honestly, I didn't think I'd make them all, and I was afraid it's to much for one PR to review 😅

I like it better when it's one PR, really.  Since the other maintainers don't know German, they can't contribute much, and therefore it all mostly looks the same to us, and it being in one PR makes it easier to comprehend (at least for me), for that reason.

**Review by bl-ue — APPROVED** [review]
Looks good!   Thank you so, **_so_** much, @marchersimon!! 🚀

**Comment by marchersimon** [comment]
No problem! What do you think of the translation guides or the versioning system? Is any of this realizable?

**Comment by bl-ue** [comment]
I've been thinking a lot about this problem, but I haven't come up with much (other than writing a Python script that compares the and checks the dates of the commits that modify each line in the source and translation, etc.).

I just read your idea and I immediately liked it, it's really simple, and I think it would be fun to see the version number increasing over time :).  The only problem, as usual, is client support.  We could perhaps put in a comment, below the title, e.g.:
```markdown
# page

<!-- version: 1 -->

> Description.
```
and clients that use a CommonMark-conformant markdown parser would ignore those.  But I'm worrried that some clients use their own simple, makeshift parsing 🤔 

But I definitely think we should discuss this idea with the others. 👍🏻  Personally, I love it! :rocket:

**Comment by sbrl** [comment]
> Language-specific translation guides

Good idea! I suspect a better place for such guides would be alongside `CONTRIBUTING.md` though.

> Page versioning system

That sounds complex and cumbersome on a number of levels. Git is a version control system, so incrementing a per-page version number in the filename would defeat the object there a bit. Also, it would complicate the implementation for clients.

> Pages with same name

As mentioned in our [client spec](https://github.com/tldr-pages/tldr/blob/master/CLIENT-SPECIFICATION.md#page-resolution), our page resolution algorithm means that pages in specific platform directories are tried _first_, before pages in _common_.

For example, if I am on a Windows computer and do `tldr cp`, I'll see `windows/cp`. If I do `tldr cp` on Solaris, I'll get `common/cp` because `sunos/cp` doesn't exist.

**Review by sbrl — APPROVED** [review]
Great work here though!

**Review by navarroaxel — CHANGES_REQUESTED** [review]

File: pages.de/common/ls.md
Comment (marchersimon) [reply pages.de/common/ls.md:17]: It's basically the same, I just wanted to go with the "linux" notation.

File: pages.de/common/ssh-copy-id.md
Comment (marchersimon) [reply pages.de/common/ssh-copy-id.md:7]: But isn't this basically like an email adress?

File: pages.de/common/sshpass.md
Comment (bl-ue) [reply pages.de/common/sshpass.md:15]: @navarroaxel are you sure about `Befehl` → `befehl`?  All nouns—proper and common—are capitalized in German.

File: pages.de/common/tar.md
Comment (marchersimon) [reply pages.de/common/tar.md:42]: Weird, I have no ideo where that came from

File: pages.de/linux/alpine.md
Comment (marchersimon) [reply pages.de/linux/alpine.md:16]: Usually we write a comma, before `dann` (`then`) in german

File: pages.de/common/git-status.md
Comment (marchersimon) [reply pages.de/common/git-status.md:4]: It seems a bit unnessecary, since its basically the same as the first sentence. Should we keep it anyways?

File: pages.de/common/ssh-agent.md
Comment (marchersimon) [reply pages.de/common/ssh-agent.md:5]: I have rewritten that line so it makes sense in that context.

File: pages.de/common/git-status.md
Comment (bl-ue) [reply pages.de/common/git-status.md:4]: I think it's fine to remove, so long as you remove it in the English version (and translations), and we all agree to _that_.

File: pages.de/common/ls.md
Comment (bl-ue) [reply pages.de/common/ls.md:17]: Actually, they're _not_ the same at all, especially as they get bigger (e.g. a `PiB` is **very** different than a `PB`), so I think it's important to clarify which system of units that it uses and normalize that decision throughout the pages.

Comment (navarroaxel) [reply pages.de/common/ls.md:17]: The default is `KiB` based on the man page:
```
The  SIZE argument is an integer and optional unit (example: 10K is 10*1024).  Units are K,M,G,T,P,E,Z,Y (powers of 1024) or KB,MB,... (powers of 1000).  Binary prefixes can be used, too: KiB=K, MiB=M, and so on.
```
Checked on my OS.

File: pages.de/common/ssh-agent.md
Comment (navarroaxel) [reply pages.de/common/ssh-agent.md:5]: @sbrl which is the guideline here?

File: pages.de/common/sshpass.md
Comment (navarroaxel) [reply pages.de/common/sshpass.md]: the variables should be in `snake_case` so I think the grammar could be ignored here, right?

**Review by navarroaxel — APPROVED** [review]

**Comment by bl-ue** [comment]
Also @navarroaxel will you please update these translations? (There are others, but they either don't have the units or translate the unit letters.)
- [`pages.de/common/ls.md`](https://github.com/tldr-pages/tldr/blob/master/pages.de/common/ls.md)
- [`pages.es/common/ls.md`](https://github.com/tldr-pages/tldr/blob/master/pages.es/common/ls.md)
- [`pages.hi/common/ls.md`](https://github.com/tldr-pages/tldr/blob/master/pages.hi/common/ls.md)
- [`pages.id/common/ls.md`](https://github.com/tldr-pages/tldr/blob/master/pages.id/common/ls.md)
- [`pages.it/common/ls.md`](https://github.com/tldr-pages/tldr/blob/master/pages.it/common/ls.md)
- [`pages.ta/common/ls.md`](https://github.com/tldr-pages/tldr/blob/master/pages.ta/common/ls.md)
- [`pages.pt_BR/common/ls.md`](https://github.com/tldr-pages/tldr/blob/master/pages.pt_BR/common/ls.md)

**Comment by bl-ue** [comment]
>This one's probably too hard to implement, but however. If each page had a version number, which gets increased by one, each time it is edited, **it would be so easy to see which translations are outdated**. Otherwise the only possibility we have is to manually compare the pages.

Yeah, the ease factor is a real pain when it comes to _updating_ translations.  I think we should try to add support to https://github.com/LukWebsForge/TldrProgress for comparing the Git commit dates and make the cell a warning if the latest commit modifying the English page is more recent than the commit modifying the translated pages.

**Comment by marchersimon** [comment]
@bl-ue that's brilliant. No additional complexity for this repo plus a nice graphical overview.

**Comment by bl-ue** [comment]
Before we merge this: @marchersimon if you search for `path/to/file` you may find some English versions of the text, not the German versions.  Will you please fix those?

**Comment by marchersimon** [comment]
So is it now ready to be merged?

File: pages.de/common/borg.md
Comment (navarroaxel) [inline pages.de/common/borg.md:21]:
```suggestion
`borg extract {{benutzer}}@{{host}}:{{pfad/zu/repo_verzeichnis}}::{{Montag}} {{pfad/zu/ziel_verzeichnis} --exclude '{{*.ext}}'`
```

**Review by navarroaxel — CHANGES_REQUESTED** [review]

**Review by bl-ue — APPROVED** [review]

File: pages.de/linux/apt-add-repository.md
Comment (navarroaxel) [reply pages.de/linux/apt-add-repository.md]: @bl-ue what do you think about this?

Comment (bl-ue) [reply pages.de/linux/apt-add-repository.md]: I'd say keep `repository`, because `repository_spec` may not be obvious to new users (like me.)  I have not much of an opinion, so if you feel strongly about it, we can change it. 👍🏻

**Review by navarroaxel — APPROVED** [review]

**Comment by bl-ue** [comment]
(@navarroaxel I responded to [one of your comments](https://github.com/tldr-pages/tldr/pull/5437#discussion_r598152273); if you made any more, unresolve the threads they're in because it's nearly impossible to find where you commented.)

File: pages.de/linux/apt-add-repository.md
Comment (sbrl) [reply pages.de/linux/apt-add-repository.md]: Agreed. I'd suggest keeping `repository` too.

File: pages.de/linux/alpine.md
Comment (sbrl) [reply pages.de/linux/alpine.md:16]: Let's stick with what's grammatically correct in German.

**Comment by bl-ue** [comment]
😋

**Pull request merged**
